### PR TITLE
Copy support on web

### PR DESCRIPTION
**Pull request by Nash0x7E2**
## Description

This PR adds support for copying text on Web. It modifies `selectable_text.dart` to support drag behaviour, registers `PointerDeviceKind.mouse` as a supported device type and removes the web platform check from `showToolbar` in `editable_text.dart`. The `kIsWeb` check was removed since the framework is not using native dom elements at the current time. Instead, the text is drawn directly to `Canvas` leaving users without the ability to copy text on the website. 

## Related Issues
 - https://github.com/flutter/flutter/issues/47234
  - https://github.com/flutter/flutter/issues/32246

## Tests
Modified the following tests:
   - text_field_test.dart 
   - selectable_text_test.dart 
  -  text_field_test.dart

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x ] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? No

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Nash0x7E2** [comment]
@mdebbar Looks like the test timed out

**Comment by mdebbar** [comment]
@Nash0x7E2 thanks for the PR.

I'm not sure this is the right approach we want to take. Showing Flutter's toolbar is something we deliberately stopped doing. In text fields for example, we let the browser show the native toolbar, selection handles, and context menu. I think we should do the same for SelectableText.

/cc @nturgut

**Comment by Nash0x7E2** [comment]
Hello @mdebbar, 
Definitely, this was meant as a temporary solution until the SDK adds support for copying text. At the moment, users do not have the ability to highlight and copy text.

**Comment by nturgut** [comment]
Hi, thanks for the PR.  We added support copy: https://github.com/flutter/engine/pull/15455 https://github.com/flutter/engine/pull/15675

Is there a use case that does not work for your case?

**Comment by Nash0x7E2** [comment]
Hi @nturgut, no, highlighting and copying text does not work for me. Tested on the latest master using chrome

**Comment by nturgut** [comment]
Hi Nash, yes you are right we don't have a solution for that. 

The solution should be implemented on the Web engine side.

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -1789,13 +1789,6 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   /// Returns `false` if a toolbar couldn't be shown, such as when the toolbar
   /// is already shown, or when no text selection currently exists.
   bool showToolbar() {
-    // Web is using native dom elements to enable clipboard functionality of the
-    // toolbar: copy, paste, select, cut. It might also provide additional
-    // functionality depending on the browser (such as translate). Due to this
-    // we should not show a Flutter toolbar for the editable text elements.
```
Comment (nturgut) [inline packages/flutter/lib/src/widgets/editable_text.dart]: We are not planning to show toolbar on web. 

Toolbar will be shown by mobile web browsers natively when the user is interacting with an editable text. Desktop web browsers does not show toolbar.

In other words we are mostly trying to optimize a natural interaction for all platforms (and browsers)

Comment (Nash0x7E2) [reply packages/flutter/lib/src/widgets/editable_text.dart]: Noted

**Pull request closed without merging**